src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AWS/SQS/ContextPropagation.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection.Emit;
+using System.Text;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SQS
+{
+    internal static class ContextPropagation
+    {
+        private const string SqsKey = "_datadog";
+
+        public static readonly Func<IDictionary<string, object>, string, IEnumerable<string>> HeadersGetter = ((carrier, key) =>
+         {
+             if (carrier.TryGetValue(key, out object value) && value is byte[] bytes)
+             {
+                 return new[] { Encoding.UTF8.GetString(bytes) };
+             }
+             else
+             {
+                 return Enumerable.Empty<string>();
+             }
+         });
+
+        public static void Inject<TMessageRequest>(SpanContext context, IDictionary messageAttributes)
+        {
+            /* TODO: Either use the optimized StringBuilder or decide the optimization is not worth it
+            StringBuilder sb = new();
+            sb.Append("{");
+            SpanContextPropagator.Instance.Inject(context, sb, ((sb, key, value) => sb.Append($"\"{key}\":\"{value}\",")));
+            sb.Remove(startIndex: sb.Length - 1, length: 1); // Remove trailing comma
+            sb.Append("}");
+            var resultString = sb.ToString();
+            */
```

Comment:
Is this a TODO for completion in this PR? Are there any benchmarks for this? Given it will run for every message, it's probably worth optimising if we can

Reply:
Now that I have a complete implementation, I will benchmark this. I just wanted to get it feature-complete before addressing this 😄

Reply:
Changed over to the pure StringBuilder approach in f51fb2b2562313b14ed5d68673100cd360f9f8da which does introduce some savings in execution time!